WordPress/gutenberg
# Make mid size parameter settable for Query Pagination block.

## What?
This PR adds an input to the Query Pagination block controlling the `mid_size` parameter of the frontend rendered `paginate_links()` function. Up until now, only the default of `2` for `mid_size` could be rendered with the Query Pagination block.

This is my first try to contribute to Gutenberg at all. I am also very new to JSX. I took heavy inspiration for this feature and PR from https://github.com/WordPress/gutenberg/pull/50779 which was merged just 4 days ago. Any feedback is welcome!

## Why?
Closes #45855. It would be very helpful because up until now, developers had to control the amount of pagination numbers via PHP filters or with JS.

## How?
The PR introduces a new attribute `mid_size` with a default of `2`, which is also the default of the called `paginate_links()` function in the frontend. The input is only shown if a Query Page Numbers block is present inside the Query Pagination block.

## Testing Instructions
1. In the editor, insert a Query Pagination block.
2. See that in the editor the default `mid_size` of `2` is applied. The Query Page Numbers block should read: `1 2 3 4 5 … 6`.
3. The number of pages in the Query Page Numbers block should decrease or increase depending on the set `mid_size`. For example, a `mid_size` of `0` should result in `1 … 2` and a `mid_size` of `4` should result in `1 2 3 4 5 6 7 8 9 … 10`.
4. Ensure that the max settable size is `10`.
5. The `mid_size` control should only be visible if a Query Page Numbers block is present. Delete the Query Page Numbers block and see, that the `mid_size` control is gone on the Query Pagination block.

### Testing Instructions for Keyboard
1. In the editor, insert a Query Pagination block.
2. Tab to the Query Pagination block settings panel.
3. Tab to the `mid_size` control and change the number using the arrow-up and arrow-down keys or input a number using a number key. See, that the number of pages in the Query Page Numbers block immediately adjusts.
4. Ensure that the max settable size is `10`.
5. Tab to the Query Page Numbers block and delete it.
6. Tab back to the settings of the Query Pagination block.
7. Ensure, that the `mid_size` control is gone.

### Front end testing instructions:
1. Ensure that your queried post type has enough posts and pages, so that the desired amount of pagination pages can be shown.
2. Set the `mid_size` to a low number like `0` or `1` and ensure that the rendered number of page links in the frontend is correct.
3. Set the `mid_size` to a high number like `3` or `4` and ensure that the rendered number of page links in the frontend is correct. 
4. If there are not enough pages, the `paginate_links()` function should ensure correct behaviour.

## Screenshots or screencast
<img width="275" alt="grafik" src="https://github.com/WordPress/gutenberg/assets/10128264/0a66b5e8-2550-4c5a-8097-4d7df06e5044">
<img width="415" alt="grafik" src="https://github.com/WordPress/gutenberg/assets/10128264/1d5cac63-3762-4856-a80c-8f9f385d4e6c">

https://github.com/WordPress/gutenberg/assets/10128264/e421486a-805f-489d-a6b1-7cd6e080083a
